pkg/e2e/verify/namespace_lists.go

```diff
@@ -0,0 +1,264 @@
+package verify
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/osde2e/pkg/common/alert"
+	"github.com/openshift/osde2e/pkg/common/config"
+	"github.com/openshift/osde2e/pkg/common/helper"
+	viper "github.com/openshift/osde2e/pkg/common/concurrentviper"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	namespaceListsTestName        = "[Suite: e2e] Namespace Lists"
+	namespaceListTestPollInterval = 30 * time.Second
+	namespaceListTestPollTimeout  = 15 * time.Minute
+)
+
+func init() {
+	alert.RegisterGinkgoAlert(namespaceListsTestName, "SD-SREP", "Trevor Nierman", "sd-cicd-alerts", "sd-cicd@redhat.com", 4)
+}
+
+var _ = ginkgo.Describe(namespaceListsTestName, func() {
+	ginkgo.Context("are accurate", func() {
+		h := helper.New()
+
+		ginkgo.It("for all managed-object configMaps in the cluster", func() {
+			log.Printf("Generating & retrieving configMaps")
+			testName := fmt.Sprintf("test-namespace-lists-%s-%d-%d", time.Now().Format("20060102-150405"), time.Now().Nanosecond()/1000000, ginkgo.GinkgoParallelNode())
+			currentConfigMaps, newConfigMaps, err := generateNamespaceConfigMaps(h, testName)
+			Expect(err).ToNot(HaveOccurred(), "Error while generating and retrieving configMaps")
+
+			log.Printf("Comparing configMaps")
+			Expect(len(currentConfigMaps)).To(Equal(len(newConfigMaps)), fmt.Sprintf("Expected to generate same configmaps as configured in managed-cluster-config.\nNumber of generated ConfigMaps: %d\nNumber of configMaps in managed-cluster-config: %d", len(newConfigMaps), len(currentConfigMaps)))
+
+			for name, newList := range newConfigMaps {
+				currentList, present := currentConfigMaps[name]
+				Expect(present).To(BeTrue(), "Expected ConfigMap %s to be present in managed-cluster-config.")
+				Expect(newList).To(Equal(currentList), fmt.Sprintf("ConfigMap '%s' is out of date and needs to be regenerated in managed-cluster-config (https://github.com/openshift/managed-cluster-config/tree/master/deploy/osd-managed-resources#readme) and managed-cluster-validating-webhooks (https://github.com/openshift/managed-cluster-validating-webhooks#updating-namespace-and-service-account-list and https://github.com/openshift/managed-cluster-validating-webhooks#updating-documenation-files).\n\nmanaged-cluster-config contains:\n%s\nShould be:\n%s", name, currentList, newList))
+			}
+
+		}, float64(viper.GetFloat64(config.Tests.PollingTimeout)))
```

Comment:
This should maybe be no less than `namespaceListTestPollTimeout` given that's the minimum grace period that the test gives for the job to run.

Reply:
Yeah, agreed; the [current default](https://github.com/openshift/osde2e/blob/main/pkg/common/config/config.go#L627) is 60 (minutes, I assume). This didn't pose an issue while testing locally, but it's possible that the value could be overridden and cause the test to fail unduly. 

An alternative would be to scale it based on `namespaceListTestPollTimeout`: ie- change from `config.Tests.PollingTimeout` to `namespaceListTestPollTimeout * 4`. 

Is this more along the lines of what you were thinking @mrbarge ?

Reply:
The time unit for the polling timeout is actually seconds, but at the moment it's not actively being enforced.. which is why it hasn't caused any issues (so far I've noticed that with moving to ginkgo v2 in #936, that will eventually change). In the meantime I've been on the lookout for tests where this could pose an issue, which is what made me notice this one. :) 

E2E runs only last for a maximum of four (? I think) hours, so where possible it's good to keep things tight. I agree with your proposal to change it from `config.Tests.PollingTimeout` to something based on the `namespaceListTestPollTimeout`, definitely had similar thoughts. 60 minutes is quite generous though - could the maximum waits for job/configmap deletion be reduced down from their current 15 minutes? I imagine that if a configmap doesn't delete within a minute or two the situation on-cluster is such that it's not going to delete in 15 either.

It's not a huge dealbreaker for me (esp. given the polling timeout is not currently enforced) but I'm just on the lookout for good test hygiene. :)

Reply:
Makes sense, thanks for the info!

I updated the test to timeout after 30 minutes (`namespaceListTestPollTimeout.Seconds()*2`). The `make-all-managed-lists.sh` script can take awhile to run, so I was trying to keep the timeout generous for those particular operations, but if you think that might be a little too long I can certainly tweak things further.